### PR TITLE
[Merged by Bors] - feat(OperatorNorm,Multilinear): add lemmas, golf

### PR DESCRIPTION
**Pull request by urkud**
### `Analysis/NormedSpace/OperatorNorm`

* Add `ContinuousLinearMap.isLeast_op_norm`, `ContinuousLinearMap.op_nnnorm_le_iff`,
  and `ContinuousLinearMap.isLeast_op_nnnorm`.
* Golf some proofs.
* Drop `RingHomIsometric` assumption here and there.

### `Analysis/NormedSpace/Multilinear`

* Add `ContinuousMultilinearMap.isLeast_op_norm`, `ContinuousMultilinearMap.isLeast_op_nnnorm`, `ContinuousMultilinearMap.le_mul_prod_of_le_op_norm_of_le`, `ContinuousMultilinearMap.op_norm_le_iff`, `ContinuousMultilinearMap.op_nnnorm_le_iff`, `ContinuousMultilinearMap.op_nnnorm_prod`, `ContinuousMultilinearMap.op_nnnorm_pi`.
* Golf/move some proofs.
* Change implicit/explicit arguments in some lemmas.

---

- [ ] depends on: #7301

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Multilinear.lean [hunk 1]
```diff
@@ -330,25 +330,19 @@ theorem bounds_bddBelow {f : ContinuousMultilinearMap 𝕜 E G} :
   ⟨0, fun _ ⟨hn, _⟩ => hn⟩
 #align continuous_multilinear_map.bounds_bdd_below ContinuousMultilinearMap.bounds_bddBelow
 
+theorem isLeast_op_norm : IsLeast {c : ℝ | 0 ≤ c ∧ ∀ m, ‖f m‖ ≤ c * ∏ i, ‖m i‖} ‖f‖ := by
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/Multilinear.lean:333]: Can you add the `nnnorm` version too, which has a slightly nicer statement? I don't have any feel for whether one is easy to prove from the other.

File: Mathlib/Analysis/NormedSpace/Multilinear.lean [hunk 2]
```diff
@@ -475,33 +468,31 @@ theorem le_of_op_nnnorm_le {C : ℝ≥0} (h : ‖f‖₊ ≤ C) : ‖f m‖₊ 
...
+  congr_arg NNReal.toReal (op_nnnorm_prod f g)
 #align continuous_multilinear_map.op_norm_prod ContinuousMultilinearMap.op_norm_prod
 
+theorem op_nnnorm_pi
+    {ι' : Type v'} [Fintype ι'] {E' : ι' → Type wE'} [∀ i', NormedAddCommGroup (E' i')]
+    [∀ i', NormedSpace 𝕜 (E' i')] (f : ∀ i', ContinuousMultilinearMap 𝕜 E (E' i')) :
+    ‖pi f‖₊ = ‖f‖₊ :=
+  eq_of_forall_ge_iff fun _ ↦ by simpa [op_nnnorm_le_iff, pi_nnnorm_le_iff] using forall_swap
+
 theorem norm_pi {ι' : Type v'} [Fintype ι'] {E' : ι' → Type wE'} [∀ i', NormedAddCommGroup (E' i')]
```
Comment (ADedecker) [inline Mathlib/Analysis/NormedSpace/Multilinear.lean:492]: Could you change this to `op_norm_pi` for consistency?

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/NormedSpace/Multilinear.lean [hunk 3]
```diff
@@ -475,33 +468,31 @@ theorem le_of_op_nnnorm_le {C : ℝ≥0} (h : ‖f‖₊ ≤ C) : ‖f m‖₊ 
...
-    rw [pi_norm_le_iff_of_nonneg]
-    exacts [fun i => (f i).le_of_op_norm_le m (norm_le_pi_norm f i),
-      mul_nonneg (norm_nonneg f) (prod_nonneg fun _ _ => norm_nonneg _)]
-  · refine' (pi_norm_le_iff_of_nonneg (norm_nonneg _)).2 fun i => _
-    refine' op_norm_le_bound _ (norm_nonneg _) fun m => _
-    refine' le_trans _ ((pi f).le_op_norm m)
-    convert norm_le_pi_norm (fun j => f j m) i
+    ‖pi f‖ = ‖f‖ :=
+  congr_arg NNReal.toReal (op_nnnorm_pi f)
 #align continuous_multilinear_map.norm_pi ContinuousMultilinearMap.norm_pi
```
Comment (urkud) [inline Mathlib/Analysis/NormedSpace/Multilinear.lean:496]:
```suggestion
#align continuous_multilinear_map.norm_pi ContinuousMultilinearMap.op_norm_pi
```

File: Mathlib/Analysis/NormedSpace/Multilinear.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Multilinear.lean:492]:
```suggestion
theorem op_norm_pi {ι' : Type v'} [Fintype ι'] {E' : ι' → Type wE'}
    [∀ i', NormedAddCommGroup (E' i')]
```

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**